src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NpgService.kt

```diff
@@ -0,0 +1,96 @@
+package it.pagopa.ecommerce.eventdispatcher.services.v2
+
+import it.pagopa.ecommerce.commons.documents.v2.authorization.NpgTransactionGatewayAuthorizationData
+import it.pagopa.ecommerce.commons.documents.v2.authorization.TransactionGatewayAuthorizationData
+import it.pagopa.ecommerce.commons.domain.v2.pojos.BaseTransactionWithRequestedAuthorization
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.OperationDto
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.OperationResultDto
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.OperationTypeDto
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.OrderResponseDto
+import it.pagopa.ecommerce.eventdispatcher.exceptions.InvalidNPGResponseException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toMono
+
+@Service
+class NpgService(
+  private val authorizationStateRetrieverService: AuthorizationStateRetrieverService
+) {
+
+  private val logger: Logger = LoggerFactory.getLogger(javaClass)
+
+  fun getAuthorizationDataFromNpgOrder(
+    transaction: BaseTransactionWithRequestedAuthorization
+  ): Mono<TransactionGatewayAuthorizationData> {
+    return getNpgOrderStatus(transaction, authorizationStateRetrieverService).flatMap {
+      when (it) {
+        is NpgOrderRefunded -> {
+          logger.info(
+            "Transaction with id [{}] in refund state, doing nothing", transaction.transactionId)
+          Mono.empty()
+        }
+        is NgpOrderAuthorized ->
+          it.authorization.operationId?.let { operationId ->
+            NpgTransactionGatewayAuthorizationData(
+                it.authorization.operationResult,
+                operationId,
+                it.authorization.paymentEndToEndId,
+                null,
+                null)
+              .toMono()
+          }
+            ?: Mono.error(InvalidNPGResponseException())
+        is UnknownNpgOrderStatus ->
```

Comment:
If we cannot identify the status of the order and no operations have been given, then we throw an exception for the DLQ, otherwise we ignore the refund

Reply:
in this case the span attribute associated to this dead letter queue will be `PROCESSING_ERROR` by the `runTracedPipelineWithDeadLetterQueue` function.
Here maybe will be better to remap this specific case to `REFUND_MANUAL_CHECK_REQUIRED`

Reply:
Never mind, I forgot the InvalidNPGResponseException to DQL event category mapping done [here](https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/blob/main/src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/common.kt#L885C1-L886) :D